pyatlan/client/asset.py

```diff
@@ -1848,14 +1876,18 @@ def _get_next_page(self):
         query = self._criteria.dsl.query
         self._criteria.dsl.size = self._size
         self._criteria.dsl.from_ = self._start
+        is_bulk_search = (
+            self._bulk or self._approximate_count > self._MASS_EXTRACT_THRESHOLD
+        )
 
-        if self._bulk:
+        if is_bulk_search:
             LOGGER.debug(
-                "Bulk search option is enabled. Ignoring requests for default offset-based "
-                "paging and switching to a creation timestamp-based paging approach."
+                self._get_bulk_search_log_message(),
```

Comment:
I'm not sure we should log this on every single page — seems unnecessary (once the decision on approach is made for the first page, it'll remain the same for all subsequent pages, right?) — can we log this only once (up-front) when the decision on paging approach is made rather than on every single page (will flood logs with repetitive messages)?